src/foam/dao/FileJournal.js

```diff
@@ -298,21 +298,17 @@ foam.CLASS({
       javaCode: `
         try {
           String line = reader.readLine();
-          if ( line == null )
-            return null;
-          if ( getMultiLine() ) {
-            StringBuilder sb = new StringBuilder();
-            while ( ! line.equals("})") ) {
-              sb.append(line);
-              line = reader.readLine();
-              if ( line == null )
-                return null;
-            }
+          if ( ! line.equals("p({") && ! line.equals("r({") ) return line;
```

Comment:
If we were really clever we could remove the other condition and just collect lines until we find the next one starting with "p({" or "r({" but that would be a bit tricky to implement as we'd have to buffer lines locally or make a PushbackBufferedReader or something.

Not something to do now, but just a thought I had while reading.